perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -893,11 +893,19 @@ sub report_edit : Path('report_edit') : Args(1) {
 
         $c->forward( '/admin/report_edit_category', [ $problem ] );
 
-        my $email = lc $c->get_param('email');
-        if ( $email ne $problem->user->email ) {
-            my $user = $c->model('DB::User')->find_or_create({ email => $email });
-            $user->insert unless $user->in_storage;
-            $problem->user( $user );
+        my $username = lc $c->get_param('username');
+        my $type = 'email';
+        my $phone_obj = FixMyStreet::SMS->parse_number($username);
+        if ($phone_obj) {
+            $type = 'phone';
+            ($username = $phone_obj->format) =~ s/\s+//g;
+        }
```

Comment:
This seems like a place where there's a chance of ending up with phone numbers in email columns.

Reply:
Hmm, yes, why isn't that calling is_valid_email anywhere...